### PR TITLE
Prioruty queue is implemented

### PR DESCRIPTION
**Pull request by vbre**

### CODE REVIEW

File: Encapsulation/Valeriya/MyExceptionClass.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Encapsulation.Valeriya
+{
+    class MyExceptionClass: Exception
+    {
+        public MyExceptionClass() : base ()
+        {
+            Console.WriteLine("There is no element with such priority");
```
Comment (jane-u) [inline Encapsulation/Valeriya/MyExceptionClass.cs]: should be corrected. discussed at class training.

File: Encapsulation/Valeriya/PriorityQueue.cs [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+                throw new IndexOutOfRangeException("The queue is empty");
+            }
+              
+            return result;
+        }
+
+        public void Enqueue(T val, int priority)
+        {
+            int index = 0;
+            if (priorityQueue.Count == 0)
```
Comment (jane-u) [inline Encapsulation/Valeriya/PriorityQueue.cs]: this seems to be unnecessary

File: Encapsulation/Valeriya/PriorityQueue.cs [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+                    break;
+                }
+            }
+
+            return result;
+        }
+
+        Tuple<int, T> GetHighestPriorityElem()
+        {
+            Tuple<int, T> minElemInQueue = priorityQueue[0];
```
Comment (jane-u) [inline Encapsulation/Valeriya/PriorityQueue.cs:117]: please use Find

File: Encapsulation/Valeriya/PriorityQueue.cs [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+                if (priorityQueue[i].Item1 < minElemInQueue.Item1)
+                {
+                    minElemInQueue = priorityQueue[i];
+                }
+            }
+
+            return minElemInQueue;
+        }
+
+        public void PrintQueue ()
```
Comment (jane-u) [inline Encapsulation/Valeriya/PriorityQueue.cs]: please move printing out of this class

**Comment by vbre** [comment]
All fixed

**Pull request merged**